src/main/java/com/cinema/validations/PasswordMatchValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.cinema.validations;
+
+import com.cinema.model.dto.UserRequestDto;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordMatchValidator
+        implements ConstraintValidator<PasswordMatchConstraint, UserRequestDto> {
+    private String password;
+    private String repeatPassword;
+
+    public void initialize(PasswordMatchConstraint constraint) {
+        this.password = constraint.password();
+        this.repeatPassword = constraint.repeatPassword();
+    }
+
+    public boolean isValid(UserRequestDto userRequestDto, ConstraintValidatorContext context) {
+
+        String passwordValue = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue(password);
+        String repeatPasswordValue = (String) new BeanWrapperImpl(userRequestDto)
+                .getPropertyValue(repeatPassword);
+
+        return passwordValue == null || passwordValue.equals(repeatPasswordValue);
```

Comment:
```suggestion
        return passwordValue != null && passwordValue.equals(repeatPasswordValue);
```
we can not be sure that @NotNull will work before our custom validation